django/db/models/options.py

```diff
@@ -992,6 +992,17 @@ def _non_pk_concrete_field_names(self):
                     names.append(field.attname)
         return frozenset(names)
 
+    @cached_property
+    def _reverse_one_to_one_field_names(self):
+        """
+        Return a set of reverse one to one field names pointing to the current model.
+        """
+        names = []
+        for field in self.related_objects:
+            if field.one_to_one:
+                names.append(field.name)
+        return frozenset(names)
```

Comment:
This can use a generator expression:

```suggestion
        return frozenset(field.name for field in self.related_objects if field.one_to_one)
```